mipengine/node/monetdb_interface/common_actions.py

```diff
@@ -16,32 +16,32 @@
 # TODO Add monetdb asyncio connection (aiopymonetdb)
 
 global_node = node_catalog.get_global_node()
-if global_node.nodeId == config.get("node", "identifier"):
+if global_node.nodeId == config.node.identifier:
     node = global_node
 else:
-    node = node_catalog.get_local_node_data(config.get("node", "identifier"))
+    node = node_catalog.get_local_node(config.node.identifier)
 monetdb_hostname = node.monetdbHostname
 monetdb_port = node.monetdbPort
 connection = pymonetdb.connect(
-    username=config.get("monet_db", "username"),
+    username=config.monetdb.username,
     port=monetdb_port,
-    password=config.get("monet_db", "password"),
+    password=config.monetdb.password,
     hostname=monetdb_hostname,
-    database=config.get("monet_db", "database"),
+    database=config.monetdb.database,
 )
 cursor = connection.cursor()
 
 
 @validate_identifier_names
 def create_table_name(
-        table_type: str, command_id: str, context_id: str, node_id: str
+    table_type: str, command_id: str, context_id: str, node_id: str
 ) -> str:
     """
     Creates a table name with the format <tableType>_<commandId>_<contextId>_<nodeId>
     """
     if table_type not in {"table", "view", "merge"}:
         raise TypeError(f"Table type is not acceptable: {table_type} .")
-    if node_id not in {"global", config.get("node", "identifier")}:
+    if node_id not in {"global", config.node.identifier}:
```

Comment:
I think we could have the 1st and the 3rd solution working together.

As you have said, we will check if there arguments are given and then we will mutate the `config` file in memory. This will work for development.

For production, with docker containers, we will not pass arguments and it will just read all of the configurations from the file.